File: binary_search.c
#include<stdio.h>
#include<stdlib.h>

int binsearch(int a[],int lb,int ub,int key){
    int mid;
    if(lb>ub)
        return -1;
    else{
        mid=(ub+lb)/2;
        if(a[mid]==key)
            return mid;
        else if(key>mid)
            return binsearch(a,mid+1,ub,key);
        else
            return binsearch(a,lb,mid-1,key);
    }
}
void main(){
    int i,n,*a,k,pos;
    printf("**********Binary Search**********\n");
    printf("Enter number of elements:");
    scanf("%d",&n);
    a=(int*)malloc(n*sizeof(int));
    printf("Enter elements\n");
    for(i=0;i<n;i++)
        scanf("%d",&a[i]);
    printf("Enter key to search: ");
    scanf("%d",&k);
    pos=binsearch(a,0,n-1,k);
    if(pos==-1)
        printf("Key not found !");
    else
        printf("%d found at position %d",k,pos+1);
}

File: python-programs-psychoder/binary_search.py
from sys import exit
def bins(a,lb,ub,key):
    mid=int((lb+ub)/2)
    if (lb>ub):
        print('key not found !')
        exit()
    else:
        if(key==a[mid]):
            print('key found at position %d'%(mid+1))
            exit()
        elif(key<a[mid]):
            bins(a,lb,mid-1,key)
        elif(key>a[mid]):
            bins(a,mid+1,ub,key)

def binary_search() :
    print('Enter the list elements:')
    a= list(map(int,input().split()))
    print('Enter key: ')
    key=int(input())
    a.sort()
    print('Sorted list:')
    print(*a)
    bins(a,0,len(a)-1,key)
print('{0:*^30}'.format('Binary Search'))
binary_search()
